packages/slice-machine/src/legacy/lib/builders/CustomTypeBuilder/SliceZone/index.tsx

```diff
@@ -203,6 +207,45 @@ const SliceZone: React.FC<SliceZoneProps> = ({
     setIsSlicesTemplatesModalOpen(false);
   };
 
+  const addSlices = async (args: {
+    slices: SharedSlice[];
+    libraryName: string;
+    mode: "ai" | "manual" | "template";
+  }) => {
+    const { slices, libraryName, mode } = args;
+
+    const serverState = await getState();
+    createSliceSuccess(serverState.libraries);
+
+    const newCustomType = addSlicesToSliceZone({
+      customType,
+      tabId,
+      slices,
+    });
+    setCustomType(CustomTypes.fromSM(newCustomType), () => {
+      toast.success(
+        <ToastMessageWithPath
+          message="Slice(s) added to slice zone and created at: "
+          path={libraryName}
+        />,
+      );
+    });
+    void completeStep("createSlice");
+
+    for (const slice of slices) {
+      void telemetry.track({
+        event: "slice:created",
+        id: slice.id,
+        name: slice.name,
+        library: libraryName,
+        location: `${customType.format}_type`,
+        mode,
+      });
+    }
+
+    syncChanges();
```

Comment:
we can sync before tracking, as it's a more important thing

Reply:
Done ☑️